theseus/src/state/children.rs

```diff
@@ -1,37 +1,159 @@
+use futures::stream::{self, StreamExt};
 use std::{collections::HashMap, sync::Arc};
-use tokio::process::Child;
+use tokio::io::{AsyncBufReadExt, BufReader};
+use tokio::process::{ChildStderr, ChildStdout};
 use tokio::sync::RwLock;
 
 // Child processes (instances of Minecraft)
-// A wrapper over a Hashmap connecting PID -> Child
-// Left open for future functionality re: polling children
-pub struct Children(HashMap<u32, Arc<RwLock<Child>>>);
+// A wrapper over a Hashmap connecting PID -> MinecraftChild
+pub struct Children(HashMap<u32, Arc<RwLock<MinecraftChild>>>);
+
+// Minecraft Child, bundles together the PID, the actual Child, and the easily queryable stdout and stderr streams
+#[derive(Debug)]
+pub struct MinecraftChild {
```

Comment:
I think we should also store the profile UUID and allow users to get a profile's running processes. In the run function we should check if the profile is already running and return an error if it is running.